### PR TITLE
Close tensorboard summary writer on finalize

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Fixes # (issue)

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tadej-redstone** [comment]
Would be great to do this for [other loggers](https://github.com/wandb/client/issues/1314) too

**Comment by borisdayma** [comment]
I was wondering about the case where you would want to log additional metrics after training (for example on test set, or other manual interpretatability metrics).

**Comment by ananthsub** [comment]
if we call close, we can replace the explicit flush: https://github.com/pytorch/pytorch/blob/master/torch/utils/tensorboard/writer.py#L1029-L1035

if we were to close, i think this should happen in on fit end, not on train end.

@borisdayma that's a good point. if you're calling trainer.fit() and then trainer.test(), the summary writer would be re-initialized. i think this would cause a version change, meaning logs from fit would go in 1 directory and logs from test into another.
